app.js

```diff
@@ -50,14 +48,45 @@ const tweet = content => { T.post('statuses/update', {status: content}, function
 
 const updateLogFile = msg => { let currentContent = fs.readFileSync(files.log, fileEncoding); currentContent += msg + os.EOL; fs.writeFileSync(files.log, currentContent);}
 
-//LOOP
+
+// SET UP
+
+// check if the twitter configuration exists
+if ((config.consumer_key == '') || (config.consumer_secret == '') || (config.access_token = '') || (config.access_token_secret == '')) {
+  // if it doesn't, check if the environment variables exist
+  if ((process.env.TWITTER_CONSUMER_KEY == undefined) || (process.env.TWITTER_CONSUMER_SECRET == undefined) || (process.env.TWITTER_ACCESS_TOKEN == undefined) || (process.env.TWITTER_ACCESS_TOKEN_SECRET == undefined)) {
+    log('> Configuration error: You have not filled in your Twitter configuration (and unable to find in environment.');
+    return false;
+  } else {
+    log('> Configuration found in environment.');
+    config.twitter_consumer_key = process.env.TWITTER_CONSUMER_KEY;
+    config.twitter_consumer_secret = process.env.TWITTER_CONSUMER_SECRET;
+    config.twitter_access_token = process.env.TWITTER_ACCESS_TOKEN;
+    config.twitter_access_token_secret = process.env.TWITTER_ACCESS_TOKEN_SECRET;
+  }
+} else if (config.isp_name == '') {
```

Comment:
I think this needs to be a separate `if`, instead of an `else if`, otherwise it will never get validated if any of the twitter config was not set.

Reply:
You are most definitely correct! Originally I didn't have the nested if statement in there, so it worked fine. D'oh!